# Hipotesis
Tienen las estudiantes de 9vo una estatura media diferente de 175 cm?
•¿Es la desviación estándar de su estatura igual a o menor que 10 cm?
•¿Es diferente la estatura promedio de las mujeres comparada con la de los hombres?

Hipótesis 1: Comparar la estatura media de las mujeres con 175 cm.

H0 (Hipótesis nula): La estatura media de las mujeres es igual a 175 cm.
H1 (Hipótesis alternativa): La estatura media de las mujeres es diferente de 175 cm.
Prueba: t-test de una muestra.
Hipótesis 2: Desviación estándar de la estatura de las mujeres.

H0: La desviación estándar de la estatura de las mujeres es igual a 10 cm.
H1: La desviación estándar de la estatura de las mujeres es menor o igual a 10 cm.
Prueba: Chi-cuadrado para varianzas.
Hipótesis 3: Comparar la estatura promedio entre hombres y mujeres.

H0: No hay diferencia en la estatura promedio entre hombres y mujeres.
H1: Hay una diferencia en la estatura promedio entre hombres y mujeres.
Prueba: t-test de dos muestras.


In [2]:
import pandas as pd

# Cargar el archivo CSV proporcionado por el usuario
file_path = 'Alturas7mo.csv'


data = pd.read_csv(file_path)

# Mostrar las primeras filas del archivo para revisar su estructura
data.head()

# Filtrar solo las filas que contienen nombres, alturas y sexo
# Eliminar filas que contienen valores no numéricos en la columna 'Altura (cm)'
cleaned_data = data[pd.to_numeric(data['Altura (cm)'], errors='coerce').notnull()]

# Seleccionar solo las columnas necesarias: Nombre, Altura (cm), Sexo
cleaned_data = cleaned_data[['Nombre', 'Altura (cm)', 'Sexo']]

# Convertir las alturas a numérico para evitar problemas
cleaned_data['Altura (cm)'] = pd.to_numeric(cleaned_data['Altura (cm)'])

# Mostrar el conjunto de datos limpio
cleaned_data.head()


Nombre  Altura (cm) Sexo
0         Fer          180    M
1         Ale          170    M
2  Alex Muñoz          170    M
3        Gael          168    M
4       Erick          165    M

Realizaremos una prueba t de una muestra para comparar la estatura promedio de las mujeres con 175 cm.
Calcularemos la prueba de chi-cuadrado para comparar la desviación estándar con 10 cm.
Haremos una prueba t de dos muestras para comparar la estatura promedio entre hombres y mujeres.

In [3]:
from scipy import stats

# Filtrar las alturas de las mujeres
women_heights = cleaned_data[cleaned_data['Sexo'] == 'F']['Altura (cm)']

# Hipótesis 1: Prueba t de una muestra (media de las mujeres comparada con 175 cm)
mu = 175  # media hipotética
t_stat, p_value = stats.ttest_1samp(women_heights, mu)

t_stat, p_value


(np.float64(-6.0), np.float64(0.10513691342250685))

Para la primera hipótesis (comparar la estatura media de las mujeres con 175 cm), obtuvimos los siguientes resultados:

t-valor: -6.0
p-valor: 0.1051
Dado que el p-valor es mayor que el nivel de significancia típico de 0.05, no tenemos suficiente evidencia para rechazar la hipótesis nula. Esto significa que no podemos concluir que la estatura media de las mujeres sea significativamente diferente de 175 cm.

Ahora, pasemos a la prueba de chi-cuadrado para comparar la desviación estándar de las mujeres con 10 cm.


In [4]:
# Hipótesis 2: Prueba de chi-cuadrado para la desviación estándar
# H0: La desviación estándar es igual a 10 cm.

# Calcular la desviación estándar muestral
std_women = women_heights.std(ddof=1)

# Número de mujeres en la muestra
n_women = len(women_heights)

# Estadístico chi-cuadrado
chi_square_stat = (n_women - 1) * (std_women ** 2) / (10 ** 2)

# p-valor asociado
p_value_chi = stats.chi2.sf(chi_square_stat, df=n_women - 1)

std_women, chi_square_stat, p_value_chi


(np.float64(1.4142135623730951),
 np.float64(0.020000000000000004),
 np.float64(0.887537083981715))

Hipótesis:
Hipótesis nula (H₀): La desviación estándar de las alturas de las mujeres es igual a 10 cm.
Hipótesis alternativa (H₁): La desviación estándar de las alturas de las mujeres es diferente de 10 cm.
Resultados obtenidos:
Desviación estándar muestral: 1.414 cm.
Estadístico chi-cuadrado: 0.02.
p-valor: 0.8875.
Interpretación:
Desviación estándar muestral (1.414 cm): Es mucho menor que la desviación estándar bajo la hipótesis nula (10 cm).

p-valor (0.8875): Este p-valor es mucho mayor que un nivel de significancia común, como 0.05. Esto significa que no tenemos suficiente evidencia para rechazar la hipótesis nula. En otras palabras, no hay evidencia significativa para concluir que la desviación estándar es diferente de 10 cm.

Conclusión:
Dado que el p-valor es muy alto (0.8875), no rechazamos la hipótesis nula. Esto indica que, con los datos disponibles, no hay suficiente evidencia para afirmar que la desviación estándar de las alturas de las mujeres es significativamente diferente de 10 cm.

En resumen, con base en estos resultados, podríamos decir que la desviación estándar observada no es significativamente distinta de 10 cm.